Fix #40: splice on select

liufengyun opened the pull request:
Fix #40: splice on select

valdisxp1 commented on src/main/scala/gestalt/Quote.scala:
```diff
@@ -183,7 +183,7 @@ abstract class Quote(val t: Toolbox, val toolboxName: String) {
       // Infix is sugar-less
       // the List is constructed as Nil.::(c).::(b).::(a)
       // Not a :: b :: c :: Nil
```
Nice, `Infix` works properly now!
The comment now is wrong, and should be removed.

liufengyun commented:
I added some helper constructors in this PR, also added `Update`, `Pat.Var`, `Pat.Unapply`, `Pat.Ascribe`, `Pat.Tuple`, `Pat.Infix`. `Ident` and `Lit` are assumed to be patterns. The types stay the same as type trees, this simplifies things while remains to be a fair assumption.

The main reason for having separate constructors for patterns is portability -- as those things are semantically very different from expressions.

But I don't think it's good to provide extractors for them, it's insane to change patterns in user-code, it's too brittle.

liufengyun commented:
I'm merging this now, I hope quasiquotes are more robust now. There are certainly more bugs, but we can fix them gradually.
